### PR TITLE
multi account integration

### PR DESCRIPTION
**Pull request by irshadCypherock**
- feat: database changes for multi account
- chore: multi account device integration

### CODE REVIEW

File: src/near/index.ts [hunk 1]
```diff
@@ -40,6 +42,18 @@ export default class NearWallet implements IWallet {
...
   }
 
+  public static getDerivationPath(addressIndex: number, _accountType: string) {
+    return (
+      intToUintByte(5, 8) +
+      intToUintByte(0x80000000 + 44, 8 * 4) +
+      intToUintByte(0x80000000 + 397, 8 * 4) +
+      intToUintByte(0x80000000, 8 * 4) +
+      intToUintByte(0x80000000, 8 * 4) +
+      intToUintByte(0x80000001 + addressIndex, 8 * 4) +
```
Comment (ghost) [inline src/near/index.ts]: Users who chose 0 as the address index for near won't be able to use their account in cysync

File: src/solana/index.ts [hunk 2]
```diff
@@ -24,25 +26,59 @@ export default class SolanaWallet implements IWallet {
...
     this.network = coin.network;
     this.address = generateSolanaAddress(this.xpub);
     this.solanaPublicKey = base_decode(this.address).toString('hex');
   }
 
+  public static getDerivationPath(accountIndex: number, _accountType: string) {
+    let derivationPath =
+      intToUintByte(0x80000000 + 44, 8 * 4) +
+      intToUintByte(0x80000000 + 501, 8 * 4);
+    let derivationDepth = 2;
```
Comment (ghost) [inline src/solana/index.ts:63]: For Solana at depth 2 let's check what ledger or at least add conditions in the database to not add for the same derivation path

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1376 Multiple accounts support in cysync - Phase 1](https://app.clickup.com/t/37308523/CSI-1376)

File: src/near/index.ts (hunk 1 above)
Comment (irshadCypherock) [reply src/near/index.ts]: Derivation path now increments as follows: 1, 0, 2, 3, 4, ...

**Review by ghost — APPROVED** [review]

**Pull request merged**